internal/gh/gh.go

```diff
@@ -26,12 +28,31 @@ func CreatePullRequest(args ...string) error {
 	return nil
 }
 
-func ensureGHInstalled() error {
+func EnsureInstalledAndAuthorized() error {
 	cmd := exec.Command("which", "gh")
 	err := cmd.Run()
 	if err != nil {
 		fmt.Println("🤓 This command requires the gh cli.\nSee: https://github.com/cli/cli")
 		return errors.New("missing gh cli dependency")
 	}
+
+	cmd = exec.Command("gh", "auth", "status")
+	output, _ := cmd.CombinedOutput()
+	outputStr := string(output)
+
+	// Check if authorized
+	tokenRegex := regexp.MustCompile(`Token: .+`)
```

Comment:
FIY, when you run `gh auth status` it returns an error code 1 if you're not authenticated. You don't need to regex the output, you can do like line 34 and check for errors. If not nil, then you're not authenticated.